packages/react/src/components/SuggestionMenu/GridSuggestionMenu/GridSuggestionMenuWrapper.tsx

```diff
@@ -0,0 +1,105 @@
+import { BlockSchema, InlineContentSchema, StyleSchema } from "@blocknote/core";
+import { FC, useCallback, useEffect } from "react";
+
+import { useBlockNoteContext } from "../../../editor/BlockNoteContext";
+import { useBlockNoteEditor } from "../../../hooks/useBlockNoteEditor";
+import { useCloseSuggestionMenuNoItems } from "../hooks/useCloseSuggestionMenuNoItems";
+import { useGridSuggestionMenuKeyboardNavigation } from "./hooks/useGridSuggestionMenuKeyboardNavigation";
+import { useLoadSuggestionMenuItems } from "../hooks/useLoadSuggestionMenuItems";
+import { GridSuggestionMenuProps } from "./types";
+
+export function GridSuggestionMenuWrapper<Item>(props: {
```

Comment:
Hmm, I see you separated both the controller, wrapper and keyboard hook. Wouldn't it suffice to just separate the controller? That's where most of the difference in implementation is. Now we introduce a lot of duplicate code

Reply:
The reason I did it this way was mostly for understandability. Basically, splitting just the controller means that you have a hierarchy like this:

- Controller (split)
- Wrapper (shared)
- Keyboard nav hook (shared)
- Actual Menu (split)

I feel like if you never saw the codebase before, it would be somewhat confusing that the suggestion menu is split between regular and grid at the top level and bottom level, but not in between. Having it fully split, while introducing more duplication, makes the code and typing simpler + (imo) makes it easier to understand without having worked on the code ever or in a long time.